document/web-api/index.bs

```diff
@@ -125,6 +135,45 @@ Note: This algorithm accepts a {{Response}} object, or a
      1. Return |returnValue|.
 </div>
 
+<h2 id="serialization">Serialization</h2>
+
+Web user agents must augment the {{Module}} interface with the <code>[<a extended-attribute>Serializable</a>]</code> extended attribute.
+
+The [=serialization steps=], given |value|, |serialized|, and |forStorage|, are:
+
+    1. If |forStorage| is true, throw a
+        "<a exception>DataCloneError</a>" {{DOMException}}
+
+    1. Set |serialized|.\[[Bytes]] to the [=sub-serialization=] of |value|.\[[Bytes]].
+
+    1. Set |serialized|.\[[AgentCluster]] to the
+       [=current Realm=]'s corresponding [=agent cluster=].
+
+The [=deserialization steps=], given |serialized| and |value|, are:
+
+    1. Let |bytes| be the [=sub-deserialization=] of |serialized|.\[[Bytes]].
+
+    1. Set |value|.\[[Bytes]] to |bytes|.
+
+    1. If |targetRealm|'s corresponding [=agent cluster=] is not
+        |serialized|.\[[AgentCluster]], then throw a
+        "<a exception>DataCloneError</a>" {{DOMException}}.
+
+    1. [=Compile a WebAssembly module=] from |bytes| and set |value|.\[[Module]] to the result.
+
+Engines should attempt to share/reuse internal compiled code when performing
+a structured serialization, although in corner cases like CPU upgrade or browser
+update, this might not be possible and full recompilation may be necessary.
+
+Note: The semantics of a structured serialization is as-if the binary source, from which the
+{{Module}} was compiled, is serialized, then deserialized, and recompiled into the target realm.
+Given the above engine optimizations, structured serialization provides developers
+explicit control over both compiled-code caching and cross-window/worker code
+sharing.
+
+Note: The web platform does not permit cross-origin sharing of serialized Modules:
+Agent clusters in HTML do not span multiple origins.
```

Comment:
This is incorrect. It does not allow sharing when two origins's hosts are not https://url.spec.whatwg.org/#host-same-site or do not have the same scheme. Otherwise sharing is possible.

Reply:
(This is also why we call it the similar-origin window agent, not just window agent.)

Reply:
I've just removed the note.